### PR TITLE
15220 update adara parser

### PR DESCRIPTION
**Pull request by peterfpeterson**
This is a version of #15253, but merging into release.

### CODE REVIEW

File: Framework/LiveData/inc/MantidLiveData/ADARA/ADARAParser.h [hunk 1]
```diff
@@ -29,6 +29,29 @@ class DLLExport Parser {
   /// Destructor
   virtual ~Parser();
 
+  long last_bytes_read;
```
Comment (peterfpeterson) [inline Framework/LiveData/inc/MantidLiveData/ADARA/ADARAParser.h]: Should this be `int64_t` and some of the others be `uint64_t`?

**Comment by peterfpeterson** [comment]
Also, the portable type that you last changed is `std::size_t`.

File: Framework/LiveData/inc/MantidLiveData/ADARA/ADARAParser.h (hunk 1 above)
Comment (rgmiller) [inline Framework/LiveData/inc/MantidLiveData/ADARA/ADARAParser.h]: Yeah, that's probably better.  I take it we actually have int64_t (and the others) on Windows?

**Comment by quantumsteve** [comment]
@peterfpeterson, Is this a good time to add `-Wcast-align` to [this](https://github.com/mantidproject/mantid/blob/master/buildconfig/CMake/DarwinSetup.cmake#L66) line  (perhaps in the branch going into master?) and fix/suppress warnings in the ADARA* files?

It is the last warning we have [enabled on GCC]( https://github.com/mantidproject/mantid/blob/master/buildconfig/CMake/GNUSetup.cmake), but not on clang. While fixing compiler warnings isn't a priority for the 3.6 release, I think it's worth going through while people are already making changes in this part of the codebase.

**Comment by AndreiSavici** [comment]
Seems to be working. I get events and log values. I will check with histograms

**Pull request merged**